lunar/distribution.yaml

```diff
@@ -2701,6 +2701,22 @@ repositories:
       url: https://github.com/ros-visualization/qwt_dependency.git
       version: kinetic-devel
     status: maintained
+  radar_omnipresense:
+    doc:
+      type: git
+      url: https://github.com/SCU-RSL-ROS/radar_omnipresense.git
+      version: 0.1.0
+    release:
+      tags:
+        release: release/lunar/{package}/{version}
+      url: https://github.com/SCU-RSL-ROS/radar_omnipresense-release.git
+      version: 0.1.0-0
+    source:
+      test_pull_requests: true
+      type: git
+      url: https://github.com/SCU-RSL-ROS/radar_omnipresense.git
+      version: 0.1.0
```

Comment:
This should be a valid Branch name. The source repository has a `master` Branch but no 0.1.0 branch (same for the source section)

Reply:
Thanks.  Hi.  So, we fastidiously followed http://wiki.ros.org/bloom/Tutorials/FirstTimeRelease  
That process promised to, and indeed did, create a tag 0.1.0 and release 0.1.0  in radar-omnipresense and  "release/lunar/radar_omnipresense/0.1.0-0" over in radar_omnipresense-release .
Because there's tag of 0.1.0 it's really difficult to create a branch named 0.1.0  (many obtuse error messages ("error: src refspec 0.1.0 matches more than one.") approximately matching what I see diagnosed in https://stackoverflow.com/questions/9378760/git-push-local-branch-with-same-name-as-remote-tag

The SO page mentions you *can* create a branch with the same name as a tag, but you don't want to because most tools fail on that (GitKraken is unhappy and SourceTree reportedly fails too) 


I did see the details below show the build error
```
======================================================================
FAIL: test_function (test.test_url_validity.TestUrlValidity)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/ros/rosdistro/test/test_url_validity.py", line 281, in test_function
    self.assertFalse(detected_errors)
AssertionError: [u"In file '''/home/travis/build/ros/rosdistro/lunar/distribution.yaml''': Could not validate source entry for repo radar_omnipresense with error [[[Could not validate repository with url https://github.com/SCU-RSL-ROS/radar_omnipresense.git and version 0.1.0 from entry at line 2714]]]"] is not false
```
if that's helpful.  
Garren and I would love any insight on how to solve this.  Thinking aloud, 
I wonder is there some way to indicate to test-url-validity that the tags created by catkin_prepare_release are sufficient?  Should something, like either the tag or branch have a "release-" suffix?  Any help would be appreciated.

Reply:
The most common approach is to have a branch like `0.1.0-devel` or `<ROSDISTRO>-devel`.

> Thinking aloud,
I wonder is there some way to indicate to test-url-validity that the tags created by catkin_prepare_release are sufficient?

This is sufficient if what you want to do is only release your package. In that case all you need is a `release` section.
The source section allows to track your development branch and to setup Continuous Integration for it. That's why a branch is required as CI will be triggered on every new commit coming on the development branch. If it was a tag or a commit hash, there would be no way to pick up new commits as they (hash and tags) are immutable.

Then it becomes a matter of if you want CI and on what branch you want the CI to run on.
If your development branch is the `master` branch you likely want to use that as the branch to test new commits and pull requests.

If you want to have a branch specifically for this release where you will push only bug fixes while continuing the development on your master branch, you can create a branch `0.1.0-release` that will be the one you use for that specific release.

Reply:
Thanks mikael.  I think our best plan for now is to forge ahead with the 0.1.0 release without a source section in distribution.yaml.  The current unit tests are not of enough value at the moment to get much out of continuous testing.  
(FYI We'll work to get it turned back on as part of the next version/checkin.  We need to add rapidjson to ros distribution first (rapidjson-dev, already present, is 8 years old and stuck at 0.1 apparently.)  Upgrading the library to modern rapidjson will allow us to deliver some more already developed features, and is a benefit to everyone.  Look for that, coming soon!)

Reply:
Sounds good, I'll wait for your commit removing the source and doc entry to review/merge this.

> rapidjson-dev, already present, is 8 years old and stuck at 0.1 apparently

Oh that's surprising, it looks like recent ubuntu had quite a version bump of this library as bionic provides 1.1.0 for `rapidjson-dev` and `rapidjson` now so hopefully that should make your life easier:
```
$ rmadison rapidjson-dev 
 rapidjson-dev | 0.12~git20141031-3      | xenial/universe | all
 rapidjson-dev | 0.12~git20141031+dfsg-1 | artful/universe | all
 rapidjson-dev | 1.1.0+dfsg2-3           | bionic/universe | all
 rapidjson-dev | 1.1.0+dfsg2-3           | cosmic/universe | all

$ rmadison rapidjson
 rapidjson | 0.12~git20141031-3      | xenial/universe | source
 rapidjson | 0.12~git20141031+dfsg-1 | artful/universe | source
 rapidjson | 1.1.0+dfsg2-3           | bionic/universe | source
 rapidjson | 1.1.0+dfsg2-3           | cosmic/universe | source
```

Reply:
Thanks for that info!  That's great, or would be if we were on 18.04  Our bots are using 16.04 on a Raspberry Pi (not unlike UbiquityRobotics ) and I fear it may be a while before there's 18.04 for a Pi3.    
 (This is a tangent from our build problems... is this the sort of question that should be asked on discourse? .... 
We were expecting we would need to add rapidjson as a source library and have the buildfarm build it as well.  If there's a deb package named rapidjson then that plan is complicated, especially since the API evolved substantially from 0.12 to 1.1.   Any tips?  (Any place we should take this conversation?  Thanks!)

Reply:
As this is a different discussion, I encourage you to ask the question on ROS Answers to get feedback on this